src/components/IframeReport.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { getContextPath } from '../utils/api'
+
+const LOADING_DONE = 'DHIS2_REPORTS_APP_IFRAME_CONTENT_LOADING_DONE'
+const ADDITIONAL_STYLE_RULES = `
+body {
+    font-family: 'Roboto', sans-serif;
+}
+table {
+    border-collapse: collapse;
+}
+th, td {
+    border: 1px solid #bcbcbc;
+    padding: 4px;
+}
+`
+
+class IframeReport extends React.PureComponent {
+    constructor(props) {
+        super(props)
+        this.state = {
+            height: 0,
+        }
+        this.iframeRef = React.createRef()
+        this.onIframeContentLoaded = this.onIframeContentLoaded.bind(this)
+        this.createScriptTag = this.createScriptTag.bind(this)
+    }
+
+    componentDidMount() {
+        this.initIframe()
+        window.addEventListener('message', this.onIframeContentLoaded)
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('message', this.onIframeContentLoaded)
+    }
+
+    initIframe = () => {
+        const iframeDoc = this.iframeRef.current.contentWindow.document
+        iframeDoc.head.innerHTML = this.createHeadHtml()
+        iframeDoc.body.innerHTML = this.createBodyHtml()
+    }
+
+    onIframeContentLoaded(event) {
+        if (event.data === LOADING_DONE) {
+            this.setHeight()
+        }
+    }
+
+    createHeadHtml() {
+        return `
+            <meta charset="utf-8">
+            ${this.props.scripts.map(this.createScriptTag).join('\n')}
+            ${this.createStyleTag()}
+        `
+    }
+
+    createScriptTag(script) {
+        const src = getContextPath() + script
+        return `<script src="${src}" type="text/javascript"></script>`
+    }
+
+    createStyleTag() {
+        return `
+        <style>
+            html {
+                overflow-y: hidden;
+            }
+            ${this.props.withStyle ? ADDITIONAL_STYLE_RULES : ''}
+        </style>
+        `
+    }
+
+    createBodyHtml() {
+        const { content } = this.props
+        /*
+         * This is where the magic happens: this image is the last
+         * child of the body, so in theory it should load last.
```

Comment:
This is wrong. Images load in parallel, so you have no control over which images has been loaded the last.
Here's a way how you could manipulate the html in order to know that (If you want to go that way): http://www.cryer.co.uk/resources/javascript/script3.htm

Reply:
Thanks for pointing that out, I was afraid of that....

I saw this approach on Stackoverflow and was pretty sceptical about it, for the reason you just mentioned. But when I tested it, to my surprise it did seem to work. Upon reflection, I made a silly mistake: the report called "Immunization: Overview (HTML-based with SVG)" has an image, so I thought it was a suitable test case. But now I see that this is actually an inline SVG, so there is nothing to load.

I just did a small test, by adding a huge image to the iframe body and throttling my network speed. This solution completely falls apart, the height is completely incorrect, and even worse, because I had put `overflow-y: hidden;` on the iframe html, you can't even scroll down in the iframe.

Thanks for sharing that link with info on how we could control the load order. In our case we could theoretically use that, but we'd have to also add yet another hacky step before that, in which we would manipulate the HTML string so that images don't have a `scr` attribute anymore (i.e. replace `src` with `data-src`, etc...).

Or alternatively, we could load some script that get all images in the DOM, and keeps track of stuff that way.

But I think we should stop here, and just pull the plug on this approach... The current solution is "interesting" already, and adding more quirkiness would make it "eccentric".... Also, we should not forget that these HTML templates may include interactivity (JS) themselves. So we don't know what will happen to a page after it has been loaded. Perhaps the page height will change due some interaction. Our our scripts start interfering with scripts in the template....